i2c-i801.c: Skip SPD initialization on cRIO-903x

billpittman opened the pull request:
The cRIO-903x architecture is sort of special, we do not connect
a dimm to the SPD because we use flash instead, but the SPD is still
present because the processor expects it.  However, enumerating and
registering the SPD i2c bus is causing interrupt storms, so for now
we skip the registration on all 903x devices.

Fixes: 01590f361e94 ("i2c: i801: Instantiate SPD EEPROMs automatically")
Signed-off-by: Bill Pittman <bill.pittman@ni.com>
Natinst-AZDO-ID: 1573148

billpittman commented:
@gratian

bstreiff commented:
Would it be better to identify 903x-type devices instead of doing this revert? This function is already doing machine identification (`if (dmi_name_in_vendors("FUJITSU")`) so adding a new machine-specific check here doesn't seem too out-of-place.

Commit:
i2c-i801.c: Skip SPD initialization on cRIO-903x

The cRIO-903x architecture is sort of special, we do not connect
a dimm to the SPD because we use flash instead, but the SPD is still
present because the processor expects it.  However, enumerating and
registering the SPD i2c bus is causing interrupt storms, so for now
we skip the registration on all 903x devices.

Fixes: 01590f3 ("i2c: i801: Instantiate SPD EEPROMs automatically")
Signed-off-by: Bill Pittman <bill.pittman@ni.com>
Natinst-AZDO-ID: 1573148

billpittman commented:
Good idea @bstreiff.
